### PR TITLE
[Blog app] Milestone 10 - Authentication with Devise

### PR DESCRIPTION
**Pull request by cherelemma**
##  Hello Dear CR ✋ 

I hope this message finds you well. I am submitting a pull request for your review and feedback on the implementation of the **Blog app - Authentication with Devise**. I, @cherelemma, and @misalazeem collaborate to implement the project requirements.  We believe the implementation fulfills the project requirements and also maintains the general requirements.

## Project Requirements ✔️

- Remove the `current_user` method in the `ApplicationController`, the devise will provide us with one.
- [Install and set up devise](https://github.com/heartcombo/devise#getting-started).
  - Can register a new user.
  - User logs in with a combination of email and password.
  - A hashed password should be stored in the database.
  - Ask for confirmation of email.
  - Can reset password.
- [Modify the views in devise](https://github.com/heartcombo/devise#configuring-views) for registration and login to match [the wireframes from the Sneak Peek](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md#:~:text=For%20this%20project%20you%20will%20have%20full%20freedom%20in%20terms%20of%20visual%20design%20but%20you%20will%20need%20to%20keep%20the%20following%20wireframes%3A) and your styling.

## Implementation Summary ✔️

In this milestone, we completed the following tasks to implement the project requirements.

➡️Install devise gem
➡️Generate devise model and update the schema
➡️Add authentication configuration to the User model and the Application controller
➡️Generate devise Views 
➡️Add style for devise Views
➡️Update routes

## General requirements ✔️

- There are [No linter errors](https://github.com/microverseinc/linters-config)
- Follow correct [GitFlow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md)
- Follow the list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md).

We have thoroughly reviewed the project requirements and cross-checked the implementation of the authentication of the Blog app against the given description to ensure accuracy. All necessary views have been run successfully.

We'd like to ask for a review, as your feedback and suggestions are invaluable in ensuring the quality and efficiency of the project. 

Thank you in advance for your time and assistance. We look forward to your feedback and addressing any concerns to enhance the overall quality of the project.

### CODE REVIEW

File: app/views/devise/sessions/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,32 @@
+<div class="auth_pages">
+  <h2>Log in</h2>
+
+  <%= form_for(resource, as: resource_name, url: session_path(resource_name)) do |f| %>
+    <div class="field">
+      <%= f.label :email %><br />
+      <%= f.email_field :email, autofocus: true, autocomplete: "email", placeholder: "Email address" %>
+    </div>
+
```
Comment (TadesseDev) [inline app/views/devise/sessions/new.html.erb:9]: - [ ] Dear Team, You have done an impressive job with the functionality, I'm only concerned your design for the login and sign-up does not match the wireframe at this point. Remember, the views in devise for the registration and login are expected to be modified by the student to match the wireframes and their styling. As you can see in the airframe the login is at the center of the page the buttons are aligned equally with the input fields and the title `Log in` is not displayed.

### Your design 
Screen Shot 2023-08-30 at 8 29 34 PM
### The wireframe

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi Team

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
212094426-b41dd457-c9db-4ab3-bbe9-176dd8dd08b7




## Highlights
 - [x] You have written a clean code ✅
 - [x]  Most of the requirement has been meet. ✅
 - [x]  functional `registration` and `login` pages ✅
 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/assets/stylesheets/application.css [hunk 2]
```diff
@@ -12,6 +12,7 @@ body {
   margin: auto;
 }
 
+.wrapper,
```
Comment (Felix45) [inline app/assets/stylesheets/application.css:15]: - [ Good Job ] Good job styling the `Login` and `Registration` forms.

Screenshot from 2023-08-30 21-05-54

**Review by Felix45 — APPROVED** [review]
# Approved :green_circle: 
Hi @cherelemma & @misalazeem , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights :heavy_check_mark: 
- [x] Good UI design :white_check_mark: 
- [x] Devise `gem` is correctly added to the project :white_check_mark: 
- [x] Git flow is used correctly :white_check_mark: 
- [x] Pull request has a descriptive summary and title :white_check_mark:   
- [x] Implemented all the changes from the previous reviewer :white_check_mark:   

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**